packages/runtime/runtime-utils/src/utils.ts

```diff
@@ -27,3 +28,43 @@ export async function seqFromTree(
 	const attrib = await readAndParseBlob<IDocumentAttributes>(attributesHash);
 	return attrib.sequenceNumber;
 }
+
+/**
+ * Encode compact ID (returned by IContainerRuntime.generateDocumentUniqueId()) to a compact string representation.
+ * While this is the main usage pattern, it works with any non-negative integer or a string.
+ * Strings are retured as is, and assumed to be UUIDs, i.e. unique enough to never overlap with
+ * numbers encoded as strings by this function. Any other strings are likely to run into collisions and should not be used!
+ * This function is useful in places where we serialize resulting ID as string and use them as strings, thus we are not
+ * gaining any efficiency from having a number type.
+ * We do not provide a decode function, so this API is only useful only result is stored and there is no need to go back to origianl form.
+ * @param numArg - input number, non-negative integer.
+ * @param prefix - optinal string prefix
+ * @returns A string - representation of an input
+ * @internal
+ */
+export function encodeCompactIdToString(numArg: number | string, prefix = "") {
+	if (typeof numArg === "string") {
+		return numArg;
+	}
+	// WARNING: result of this function are encodeCompactIdToStringn storage!
+	// If you ever need to change this function, you will need to ensure that
+	// for any inputs N1 & N2, old(N1) !== new(N2), where old() - is the old implementation,
+	// and new() - is new implementation of encodeCompactIdToString()
+	// This likely means - this function can't be changed, unless it uses some prefix that ensures
+	// new values have zero overlap with old values.
+	// Also resulting string can't container "/", as that's disallowed by some users
+	// (data store and DDS IDs can't have "/" in their IDs).
+	assert(Number.isInteger(numArg) && numArg >= 0, "invalid input");
+	let id = "";
+	let num = numArg;
+	do {
+		// 48-57 -> 0-9
+		// 65-91 > A-Z[
+		// 97-123 -> a-z}
+		const encode = num % 64;
```

Comment:
Is this a standard encoding technique? It's hard for me to tell what kind of encoding it's doing and how to be sure it will generate unique ids.

Reply:
Maybe add some comments around the encoding algorithm. You can also add examples to make it clear.

Reply:
No, it's not standard if by standard you mean base64 encoding charset. They use "/" and we can't use "/".
Plus, base64 is for encoding mostly binary content - if you give it an integer in string form, it will take 33% more space, not less :) So I'm not sure how valuable it is here (in addition to be relative slow).
Will add some examples....